This notebook presents examples of the prodest package by Gabriele Rovigatti. The examples are taken from the [package manual](https://cran.r-project.org/web/packages/prodest/prodest.pdf).

In [64]:
require(prodest)

# Printing Latex
To print the $\LaTeX$ output from the prodest `printProd` command in Jupyter Notebooks you need to paste it into a markdown cell and modify it slightly to satisfy MathJax rendering. The only modificiation needed is to replace `tabular` with `array`.

# panelSim

In [65]:
## Simulate a dataset with 100 firms (N=100) over 50 time periods (T = 50).
## \code{Panelsim()} delivers the last 10% of usable time per panel.
panel.data <- panelSim(N = 100, T = 50)
attach(panel.data)

## Estimate various models
ACF.fit <- prodestACF(Y, fX, sX, pX2, idvar, timevar, theta0 = c(.5,.5))
LP.fit <- prodestLP(Y, fX, sX, pX2, idvar, timevar)
WRDG.fit <- prodestWRDG(Y, fX, sX, pX3, idvar, timevar, R = 5)

## print results in lateX tabular format
printProd(list(LP.fit, ACF.fit, WRDG.fit))

The following objects are masked from panel.data (pos = 3):

    fX, idvar, pX1, pX2, pX3, pX4, sX, timevar, Y

The following objects are masked from panel.data (pos = 4):

    fX, idvar, pX1, pX2, pX3, pX4, sX, timevar, Y

The following objects are masked from panel.data (pos = 6):

    fX, idvar, pX1, pX2, pX3, pX4, sX, timevar, Y



\begin{tabular}{ccccccc}\hline\hline
 & & LP & & ACF & & WRDG \\\hline
 fX1 & & 0.656 & & 0.443 & & 0.835 \\
 & & (0.021) & & (0.043) & & (0.023) \\
 &  &  &  \\
 sX1 & & 0.382 & & 0.583 & & 0.168 \\
 & & (0.037) & & (0.086) & & (0.184) \\
 &  &  &  \\
 &  &  &  \\
N & & 500 & & 500 & & 500 \\\hline\hline
\end{tabular}

$$
\begin{array}{ccccccc}\hline\hline
 & & LP & & ACF & & WRDG \\\hline
 fX1 & & 0.656 & & 0.443 & & 0.835 \\
 & & (0.021) & & (0.043) & & (0.023) \\
 &  &  &  \\
 sX1 & & 0.382 & & 0.583 & & 0.168 \\
 & & (0.037) & & (0.086) & & (0.184) \\
 &  &  &  \\
 &  &  &  \\
N & & 500 & & 500 & & 500 \\\hline\hline
\end{array}
$$

# Data

 Chilean data on production.The full version is publicly available at http://www.ine.cl/canales/chile_estadistico/estadisticas_economicas/industria/series_estadisticas/series_estadisticas_enia.php
 
* Y : vector of log value added
* sX : vector of log capital
* fX : matrix of log skilled labor and log unskilled labor
* pX : vector of log electricity
* cX : vector of log water
* inv : vector of log investment 
* idvar : vector of panel identifier
* timevar : vector of time

s for state. f for free. c for control. p for proxy.

In [66]:
data("chilean")
chilean[0:25,]

Y         sX        fX1      fX2       pX       cX        inv       idvar
1  10.224230  5.521461 0.000000 0.0000000 3.091043  4.491633  4.491633 10007
2  10.228210  5.525453 0.000000 0.0000000 2.995732  4.502773  4.502773 10007
3  10.230230  8.991811 0.000000 0.0000000 2.995732  8.971534  8.971534 10007
4   8.989694  9.079776 0.000000 1.6094380 1.791759  8.489083  8.489083 10007
5   8.769663  9.366232 0.000000 1.0986120 1.609438  8.914953  8.914953 10007
6  12.368220 12.364870 3.091043 0.0000000 3.850147 11.956280 11.956280 10016
7  12.404430 12.415080 3.091043 0.0000000 2.995732 11.978640 11.978640 10016
8  12.553770 12.473750 3.091043 0.0000000 3.091043 12.044080 12.044080 10016
9  12.503070 12.324800 2.995732 0.0000000 2.995732 11.766410 11.766410 10016
10 12.522590 12.456930 2.639057 0.0000000 2.944439 12.061070 12.061070 10016
11 12.355020 12.232580 2.639057 0.0000000 3.433987 11.612700 11.612700 10016
12 12.454090 12.357480 2.639057 0.0000000 3.178054 11.954460 11.954460 10016
13 12.239450 12.381590 2.484907 0.0000000 3.295837 11.930240 11.930240 10016
14 12.060680 11.679800 1.386294 2.8332130 4.836282 11.256340 11.256340 10044
15 12.060640 11.484430 1.609438 2.8332130 4.343805 10.835150 10.835150 10044
16 12.254170 12.467050 1.609438 2.5649490 4.875197 12.305280 12.305280 10044
17 12.031390 12.484890 1.386294 2.4849070 4.691348 12.020260 12.020260 10044
18 11.734500 12.193770 1.386294 0.0000000 4.477337 11.514930 11.514930 10044
19 11.487690 12.552810 1.386294 0.0000000 5.135798 12.247270 12.247270 10044
20 11.468500 10.105570 2.079442 0.6931472 1.945910  9.433647  9.433647 10075
21 11.333890 10.183310 2.079442 0.0000000 4.442651  9.651905  9.651905 10075
22 11.227030 10.243310 1.945910 0.6931472 3.178054  9.703737  9.703737 10075
23 11.056410 10.080340 2.079442 1.0986120 1.386294  9.349135  9.349135 10075
24 10.860320 10.545370 1.945910 0.6931472 1.386294 10.216280 10.216280 10075
25 11.272510 10.568830 1.791759 0.0000000 2.564949 10.023800 10.023800 10075
   timevar
1  1999   
2  2000   
3  2001   
4  2002   
5  2003   
6  1996   
7  1997   
8  1998   
9  1999   
10 2000   
11 2001   
12 2002   
13 2003   
14 1996   
15 1997   
16 1998   
17 2000   
18 2001   
19 2002   
20 1997   
21 1998   
22 1999   
23 2000   
24 2001   
25 2002

# printProd

In [67]:
# run various models
WRDGfit <- prodestWRDG_GMM(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, chilean$timevar)

OPfit <- prodestOP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, chilean$timevar)

# show the output in latex - tabular format
printProd(list(OPfit, WRDGfit), modnames = c('Olley-Pakes', 'Wooldridge'),
parnames = c('bunsk', 'bsk', 'bk'))

\begin{tabular}{ccccc}\hline\hline
 & & Olley-Pakes & & Wooldridge \\\hline
 bunsk & & 0.199 & & 0.247 \\
 & & (0.025) & & (0.017) \\
 &  &  \\
 bsk & & 0.169 & & 0.217 \\
 & & (0.022) & & (0.015) \\
 &  &  \\
 bk & & 0.117 & & 0.058 \\
 & & (0.041) & & (0.031) \\
 &  &  \\
 &  &  \\
N & & 2544 & & 1944 \\\hline\hline
\end{tabular}

$$
\begin{array}{ccccc}\hline\hline
 & & Olley-Pakes & & Wooldridge \\\hline
 bunsk & & 0.199 & & 0.247 \\
 & & (0.025) & & (0.017) \\
 &  &  \\
 bsk & & 0.169 & & 0.217 \\
 & & (0.022) & & (0.015) \\
 &  &  \\
 bk & & 0.117 & & 0.058 \\
 & & (0.041) & & (0.031) \\
 &  &  \\
 &  &  \\
N & & 2544 & & 1944 \\\hline\hline
\end{array}
$$

# prodestACF

In [68]:
# we fit a model with two free (skilled and unskilled), one state (capital)
# and one proxy variable (electricity)
ACF.fit <- prodestACF(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, chilean$timevar,
                      theta0 = c(.5,.5,.5), seed = 154673, R = 5)

# require('Rsolnp')
# There are three second-stage optization options 'optim', 'DEoptim' or 'solnp'
ACF.fit.DEoptim <- prodestACF(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, 
                            chilean$timevar, theta0 = c(.5,.5,.5), opt = 'DEoptim', seed = 154673)

In [69]:
# run the same regression in parallel
nCores <- as.numeric(Sys.getenv("NUMBER_OF_PROCESSORS"))
cl <- makeCluster(getOption("cl.cores", nCores - 1))

ACF.fit.par <- prodestACF(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, 
                          chilean$timevar, theta0 = c(.5,.5,.5), cluster = cl, seed = 154673)
stopCluster(cl)

In [70]:
# show results
coef(ACF.fit)
coef(ACF.fit.DEoptim)
coef(ACF.fit.par)

fX1       fX2       sX1 
0.6456639 0.6440158 0.2508290

fX1       fX2       sX1 
0.6049626 0.5000000 0.5000000

fX1       fX2       sX1 
0.6456639 0.6440158 0.2508290

## Productivity residuals
`omega()` generates the residuals of the second stage, namely, the estimates of the log productivity term.

In [91]:
# ACF.fit.coef <- coef(ACF.fit)
# chilean$Y - ACF.fit.coef[3]*chilean$sX - ACF.fit.coef[1]*cbind(chilean$fX1, chilean$fX2) - ACF.fit.coef[2]*chilean$pX
# chilean$Y - ACF.fit.coef[3]*chilean$sX - ACF.fit.coef[1]*chilean$fX1 - ACF.fit.coef[2]*chilean$pX

In [93]:
head(omega(ACF.fit)) #omega is a built in function that runs the above equation to get tfp estimates

[,1]    
[1,] 8.839287
[2,] 8.842266
[3,] 7.974823
[4,] 5.675719
[5,] 5.712817
[6,] 7.270977

In [73]:
# show results in .tex tabular format
printProd(list(ACF.fit, ACF.fit.DEoptim))

\begin{tabular}{ccccc}\hline\hline
 & & ACF & & ACF \\\hline
 fX1 & & 0.646 & & 0.605 \\
 & & (0.302) & & (0.174) \\
 &  &  \\
 fX2 & & 0.644 & & 0.5 \\
 & & (0.193) & & (0.126) \\
 &  &  \\
 sX1 & & 0.251 & & 0.5 \\
 & & (0.063) & & (0) \\
 &  &  \\
 &  &  \\
N & & 2544 & & 2544 \\\hline\hline
\end{tabular}

$$
\begin{array}{ccccc}\hline\hline
 & & ACF & & ACF \\\hline
 fX1 & & 0.646 & & NA \\
 & & (0.302) & & (0.169) \\
 &  &  \\
 fX2 & & 0.644 & & NA \\
 & & (0.193) & & (0.292) \\
 &  &  \\
 sX1 & & 0.251 & & NA \\
 & & (0.063) & & (0.052) \\
 &  &  \\
 &  &  \\
N & & 2544 & & 2544 \\\hline\hline
\end{array}
$$

# prodestLP

In [74]:
# we fit a model with two free (skilled and unskilled), one state (capital)
# and one proxy variable (electricity)
LP.fit <- prodestLP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX,chilean$pX, chilean$idvar, 
                    chilean$timevar, seed = 154673)

LP.fit.solnp <- prodestLP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, 
                          chilean$idvar, chilean$timevar, opt = 'solnp')

## Not run:
# run the same model in parallel
require(parallel)
nCores <- as.numeric(Sys.getenv("NUMBER_OF_PROCESSORS"))
cl <- makeCluster(getOption("cl.cores", nCores - 1))

LP.fit.par <- prodestLP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, 
                        chilean$timevar, cluster = cl, seed = 154673)
stopCluster(cl)
## End(Not run)

In [75]:
# show results
summary(LP.fit)
summary(LP.fit.solnp)

Length  Class   Mode 
     1   prod     S4 

Length  Class   Mode 
     1   prod     S4 

In [76]:
show(LP.fit)
show(LP.fit.solnp)


-------------------------------------------------------
-            Production Function Estimation           -
-------------------------------------------------------
                   Method:    LP              
-------------------------------------------------------
                         fX1       fX2       sX1 
Estimated Parameters:   0.199     0.169     0.117 
                       (0.024)   (0.024)   (0.045)
-------------------------------------------------------
-------------------------------------------------------
-            Production Function Estimation           -
-------------------------------------------------------
                   Method:    LP              
-------------------------------------------------------
                         fX1       fX2       sX1 
Estimated Parameters:   0.199     0.169     0.117 
                       (0.025)   (0.022)   (0.041)
-------------------------------------------------------

In [77]:
# show results in .tex tabular format
printProd(list(LP.fit, LP.fit.solnp))

\begin{tabular}{ccccc}\hline\hline
 & & LP & & LP \\\hline
 fX1 & & 0.199 & & 0.199 \\
 & & (0.024) & & (0.025) \\
 &  &  \\
 fX2 & & 0.169 & & 0.169 \\
 & & (0.024) & & (0.022) \\
 &  &  \\
 sX1 & & 0.117 & & 0.117 \\
 & & (0.045) & & (0.041) \\
 &  &  \\
 &  &  \\
N & & 2544 & & 2544 \\\hline\hline
\end{tabular}

$$
\begin{array}{ccccc}\hline\hline
 & & LP & & LP \\\hline
 fX1 & & 0.199 & & 0.199 \\
 & & (0.024) & & (0.025) \\
 &  &  \\
 fX2 & & 0.169 & & 0.169 \\
 & & (0.024) & & (0.022) \\
 &  &  \\
 sX1 & & 0.117 & & 0.117 \\
 & & (0.045) & & (0.041) \\
 &  &  \\
 &  &  \\
N & & 2544 & & 2544 \\\hline\hline
\end{array}
$$

# prodestOP

In [78]:
# we fit a model with two free (skilled and unskilled), one state (capital)
# and one proxy variable (electricity)
OP.fit <- prodestOP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX,
chilean$inv, chilean$idvar, chilean$timevar)
OP.fit.solnp <- prodestOP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2),
chilean$sX, chilean$inv, chilean$idvar,
chilean$timevar, opt='solnp')
OP.fit.control <- prodestOP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2),
chilean$sX, chilean$inv, chilean$idvar,
chilean$timevar, cX = chilean$cX)

In [79]:
# show results
summary(OP.fit)
summary(OP.fit.solnp)
summary(OP.fit.control)

Length  Class   Mode 
     1   prod     S4 

Length  Class   Mode 
     1   prod     S4 

Length  Class   Mode 
     1   prod     S4 

In [80]:
# show results in .tex tabular format
printProd(list(OP.fit, OP.fit.solnp, OP.fit.control))

\begin{tabular}{ccccccc}\hline\hline
 & & OP & & OP & & OP \\\hline
 fX1 & & 0.314 & & 0.314 & & 0.314 \\
 & & (0.034) & & (0.034) & & (0.034) \\
 &  &  &  \\
 fX2 & & 0.256 & & 0.256 & & 0.256 \\
 & & (0.036) & & (0.036) & & (0.036) \\
 &  &  &  \\
 sX1 & & 0.168 & & 0.165 & & 0.168 \\
 & & (0.028) & & (0.03) & & (0.028) \\
 &  &  &  \\
 &  &  &  \\
N & & 2544 & & 2544 & & 2544 \\\hline\hline
\end{tabular}

\begin{array}{ccccccc}\hline\hline
 & & OP & & OP & & OP \\\hline
 fX1 & & 0.314 & & 0.314 & & 0.314 \\
 & & (0.034) & & (0.034) & & (0.034) \\
 &  &  &  \\
 fX2 & & 0.256 & & 0.256 & & 0.256 \\
 & & (0.036) & & (0.036) & & (0.036) \\
 &  &  &  \\
 sX1 & & 0.168 & & 0.165 & & 0.168 \\
 & & (0.028) & & (0.03) & & (0.028) \\
 &  &  &  \\
 &  &  &  \\
N & & 2544 & & 2544 & & 2544 \\\hline\hline
\end{array}

# prodestWRDG

In [81]:
# we fit a model with one free (unskilled), one state (capital)
# and one proxy variable (electricity)
WRDG.fit <- prodestWRDG(chilean$Y, chilean$fX1, chilean$sX, chilean$pX, chilean$idvar, chilean$timevar)
# show results
WRDG.fit


-------------------------------------------------------
-            Production Function Estimation           -
-------------------------------------------------------
                   Method:    WRDG              
-------------------------------------------------------
                         fX1       sX1 
Estimated Parameters:   0.092     0.055 
                       (0.025)   (0.078)
-------------------------------------------------------

In [82]:
# Takes a long time to run

# estimate a panel dataset - DGP1, various measurement errors - and run the estimation
sim <- panelSim()
WRDG.sim1 <- prodestWRDG(sim$Y, sim$fX, sim$sX, sim$pX1, sim$idvar, sim$timevar)
WRDG.sim2 <- prodestWRDG(sim$Y, sim$fX, sim$sX, sim$pX2, sim$idvar, sim$timevar)
WRDG.sim3 <- prodestWRDG(sim$Y, sim$fX, sim$sX, sim$pX3, sim$idvar, sim$timevar)
WRDG.sim4 <- prodestWRDG(sim$Y, sim$fX, sim$sX, sim$pX4, sim$idvar, sim$timevar)

# show results in .tex tabular format
printProd(list(WRDG.sim1, WRDG.sim2, WRDG.sim3, WRDG.sim4), parnames = c('Free','State'))

\begin{tabular}{ccccccccc}\hline\hline
 & & WRDG & & WRDG & & WRDG & & WRDG \\\hline
 Free & & 0.474 & & 0.758 & & 0.84 & & 0.906 \\
 & & (0.013) & & (0.007) & & (0.006) & & (0.004) \\
 &  &  &  &  \\
 State & & 0.957 & & 0.707 & & 0.585 & & 0.422 \\
 & & (0.218) & & (0.143) & & (0.119) & & (0.089) \\
 &  &  &  &  \\
 &  &  &  &  \\
N & & 10000 & & 10000 & & 10000 & & 10000 \\\hline\hline
\end{tabular}

$$
\begin{array}{ccccccccc}\hline\hline
 & & WRDG & & WRDG & & WRDG & & WRDG \\\hline
 Free & & 0.474 & & 0.758 & & 0.84 & & 0.906 \\
 & & (0.013) & & (0.007) & & (0.006) & & (0.004) \\
 &  &  &  &  \\
 State & & 0.957 & & 0.707 & & 0.585 & & 0.422 \\
 & & (0.218) & & (0.143) & & (0.119) & & (0.089) \\
 &  &  &  &  \\
 &  &  &  &  \\
N & & 10000 & & 10000 & & 10000 & & 10000 \\\hline\hline
\end{array}
$$

# prodestWRDG_GMM

In [83]:
# we fit a model with two free (skilled and unskilled), one state (capital)
# and one proxy variable (electricity)
WRDG.GMM.fit <- prodestWRDG_GMM(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, 
                                chilean$timevar)
# show results
WRDG.GMM.fit


-------------------------------------------------------
-            Production Function Estimation           -
-------------------------------------------------------
                   Method:    WRDG              
-------------------------------------------------------
                         fX1       fX2       sX1 
Estimated Parameters:   0.247     0.217     0.058 
                       (0.017)   (0.015)   (0.031)
-------------------------------------------------------

In [84]:
# estimate a panel dataset - DGP1, various measurement errors - and run the estimation
sim <- panelSim()
WRDG.GMM.sim1 <- prodestWRDG_GMM(sim$Y, sim$fX, sim$sX, sim$pX1, sim$idvar, sim$timevar)
WRDG.GMM.sim2 <- prodestWRDG_GMM(sim$Y, sim$fX, sim$sX, sim$pX2, sim$idvar, sim$timevar)
WRDG.GMM.sim3 <- prodestWRDG_GMM(sim$Y, sim$fX, sim$sX, sim$pX3, sim$idvar, sim$timevar)
WRDG.GMM.sim4 <- prodestWRDG_GMM(sim$Y, sim$fX, sim$sX, sim$pX4, sim$idvar, sim$timevar)
# show results in .tex tabular format
printProd(list(WRDG.GMM.sim1, WRDG.GMM.sim2, WRDG.GMM.sim3, WRDG.GMM.sim4),
parnames = c('Free','State'))

\begin{tabular}{ccccccccc}\hline\hline
 & & WRDG & & WRDG & & WRDG & & WRDG \\\hline
 Free & & 0.474 & & 0.758 & & 0.84 & & 0.906 \\
 & & (0.013) & & (0.008) & & (0.007) & & (0.005) \\
 &  &  &  &  \\
 State & & 0.957 & & 0.707 & & 0.585 & & 0.422 \\
 & & (0.072) & & (0.065) & & (0.063) & & (0.06) \\
 &  &  &  &  \\
 &  &  &  &  \\
N & & 9000 & & 9000 & & 9000 & & 9000 \\\hline\hline
\end{tabular}

$$
\begin{array}{ccccccccc}\hline\hline
 & & WRDG & & WRDG & & WRDG & & WRDG \\\hline
 Free & & 0.474 & & 0.758 & & 0.84 & & 0.906 \\
 & & (0.013) & & (0.008) & & (0.007) & & (0.005) \\
 &  &  &  &  \\
 State & & 0.957 & & 0.707 & & 0.585 & & 0.422 \\
 & & (0.072) & & (0.065) & & (0.063) & & (0.06) \\
 &  &  &  &  \\
 &  &  &  &  \\
N & & 9000 & & 9000 & & 9000 & & 9000 \\\hline\hline
\end{array}
$$

# R Vignette Paper

From the R Vignette Paper. Needs many adjustments to run. 

In [85]:
# OP <- prodestOP(Y, fX, sX, chilean$inv, idvar, timevar, R = 20)
OP <- prodestOP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$inv, chilean$idvar, chilean$timevar, 
                R = 20)

# LP <- prodestLP(chilean$Y, chilean$fX, chilean$sX, chilean$pX, chilean$idvar, chilean$timevar, R = 20)
LP <- prodestLP(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, chilean$timevar, 
                R = 20)

# ACF <- prodestACF(Y, fX, sX, pX, idvar, timevar, R = 20, theta0 = (c(.5,.5,.5)))
ACF <- prodestACF(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, chilean$timevar, 
                  theta0 = c(.5,.5,.5), R=20)

# WRDG <- prodestWRDG(Y, fX, sX, pX, idvar, timevar, R = 5)
WRDG <- prodestWRDG(chilean$Y, chilean$fX1, chilean$sX, chilean$pX, chilean$idvar, chilean$timevar, R=5)

# WRDG.GMM <- prodestWRDG_GMM(Y, fX, sX, pX, idvar, timevar, R = 5)
WRDG.GMM <- prodestWRDG_GMM(chilean$Y, fX = cbind(chilean$fX1, chilean$fX2), chilean$sX, chilean$pX, chilean$idvar, 
                            chilean$timevar)

In [86]:
printProd(list(OP,LP,ACF,WRDG,WRDG.GMM), modnames = c('OP','LP','ACF','WRDG','WRDG.GMM'), 
          parnames = c('$\\beta_{skil}$','$\\beta_{unskil}$','$\\beta_{k}$'))

\begin{tabular}{ccccccccccc}\hline\hline
 & & OP & & LP & & ACF & & WRDG & & WRDG.GMM \\\hline
 $\beta_{skil}$ & & 0.314 & & 0.199 & & 0.646 & & 0.092 & & 0.247 \\
 & & (0.034) & & (0.025) & & (0.161) & & (0.015) & & (0.017) \\
 &  &  &  &  &  \\
 $\beta_{unskil}$ & & 0.256 & & 0.169 & & 0.644 & & 0.055 & & 0.217 \\
 & & (0.036) & & (0.022) & & (0.196) & & (0.118) & & (0.015) \\
 &  &  &  &  &  \\
 $\beta_{k}$ & & 0.168 & & 0.117 & & 0.251 & & NA & & 0.058 \\
 & & (0.028) & & (0.041) & & (0.033) & & (NA) & & (0.031) \\
 &  &  &  &  &  \\
 &  &  &  &  &  \\
N & & 2544 & & 2544 & & 2544 & & 2544 & & 1944 \\\hline\hline
\end{tabular}

$$
\begin{array}{ccccccccccc}\hline\hline
 & & OP & & LP & & ACF & & WRDG & & WRDG.GMM \\\hline
 \beta_{skil} & & 0.314 & & 0.199 & & 0.646 & & 0.092 & & 0.247 \\
 & & (0.034) & & (0.025) & & (0.161) & & (0.015) & & (0.017) \\
 &  &  &  &  &  \\
 \beta_{unskil} & & 0.256 & & 0.169 & & 0.644 & & 0.055 & & 0.217 \\
 & & (0.036) & & (0.022) & & (0.196) & & (0.118) & & (0.015) \\
 &  &  &  &  &  \\
 \beta_{k} & & 0.168 & & 0.117 & & 0.251 & & NA & & 0.058 \\
 & & (0.028) & & (0.041) & & (0.033) & & (NA) & & (0.031) \\
 &  &  &  &  &  \\
 &  &  &  &  &  \\
N & & 2544 & & 2544 & & 2544 & & 2544 & & 1944 \\\hline\hline
\end{array}
$$